.travis.yml

```diff
@@ -65,23 +63,38 @@ before_script:
     - if [ "$FLAVOR" == "cs" ]; then travis_retry nuget restore cs/cs.sln; fi
 
 script:
-    - mkdir build && cd build
+    - mkdir build
+
+    - if [[ "$FLAVOR" == cpp-* ]]; then
+        docker run -v $HOME/.ccache:/root/.ccache -v $HOME/.stack:/root/.stack -v `pwd`:/root/bond -i bond-xenial sh -c "
+            mkdir -p $HOME;
+            ln -s /root/.ccache $HOME/.ccache;
+            ln -s /root/.stack $HOME/.stack;
+            export BOOST_ROOT=/opt/boosts/boost_`echo $BOOST | tr . _`;
+            export CXX='ccache clang++ -Qunused-arguments --system-header-prefix=boost/';
+            export CC='ccache clang -Qunused-arguments --system-header-prefix=boost/';
+            cd /root/bond/build;
+            cmake $CPP_CMAKE_ARGS ..;
+            make --jobs 2 check;
+            ";
+      else
+        cd build;
+      fi
+
+    - if [ "$FLAVOR" == "cs" ]; then
+        cmake -DBOND_SKIP_GBC_TESTS=TRUE -DBOND_SKIP_CORE_TESTS=TRUE -DBOND_ENABLE_GRPC=FALSE ..;
+        make gbc;
+        make DESTDIR=$HOME install;
+        cd ..;
+        export BOND_COMPILER_PATH=$HOME/usr/local/bin;
+        xbuild /p:Configuration=Debug cs/cs.sln;
```

Comment:
I just merged in 85995fe which switched from `xbuild` to `msbuild` here: can you merge with the current master and fix these up?

Reply:
Merged.